File: src/main/java/com/company/commitet_jm/entity/TypeStorage.java
package com.company.commitet_jm.entity;

import io.jmix.core.metamodel.datatype.EnumClass;

public enum TypeStorage implements EnumClass<Integer> {
    CF(10),
    CFE(20);

    private final Integer id;

    TypeStorage(Integer id) {
        this.id = id;
    }

    @Override
    public Integer getId() {
        return id;
    }

    public static TypeStorage fromId(Integer id) {
        for (TypeStorage type : TypeStorage.values()) {
            if (type.id.equals(id)) {
                return type;
            }
        }
        return null;
    }
}

File: src/main/kotlin/com/company/commitet_jm/entity/TypeStorage.kt
package com.company.commitet_jm.entity

import io.jmix.core.metamodel.datatype.EnumClass

enum class TypeStorage(private val id: Int) : EnumClass<Int> {
    CF(10),
    CFE(20);

    override fun getId() = id

    companion object {

        @JvmStatic
        fun fromId(id: Int): TypeStorage? = TypeStorage.values().find { it.id == id }
    }
}